### ISSUE TITLE
Rule to disallow duplicate keys within block

### ISSUE THREAD

**Issue by jasonmobley** [tclindner/npm-package-json-lint#171]
Would be nice to have a rule that errors or warns if the same key is defined twice in the same block. My use case is really making sure a _dependency_ is not listed twice, but the idea applies across the whole file.

For example, this is probably a mistake:
```json
{
  "name": "my-cool-package",
  "version": "1.2.3",
  "version": "4.5.6"
}
```

This may be easier said than done since the behavior of `JSON.parse()` appears to be "use the last value defined for that key as its parsed value."

**Comment by tclindner** [comment]
Thank you, @jasonmobley. I like the idea.

**Comment by tclindner** [comment]
Released in v4.5.0. Thank you, @effrenus! 🎉
